### DISCUSSION TITLE
healthchecks.io like service support to monitor cron jobs etc

### DISCUSSION THREAD

**Discussion by yarikoptic** [upptime/upptime#986] [Ideas]
There is a https://healthchecks.io/ which works by cron jobs `curl`ing a dedicated to the cron job endpoint upon successful completion. This allows to ensure that CRON jobs across infrastructure do their job and not disabled or stuck.

I wondered if we could somehow add similar checks to upptime -- that would have been awesome to centralize overview of the health status of the systems.

Just mindstorming ideas
- if to follow original https://healthchecks.io/ model - it would require some service with logging valid (to avoid DDoS etc) requests, "privately" (so that UUIDs are not leaked for DDoS/abuse etc)
- for a private github repo, some metadata field (edit timestamp or better an explicit content within issue for a timestamp or may be even some stats record: number of pings, earliest ping, missed pings, etc.) on an issue could be used. Then it could be a private `uptime-pings` repo, and accompany with some little shell script helper to "load token", ping corresponding issue.  Then upptime itself would just get stats from those issues. Each service could be configured within that issue record (on how frequently to expect pings etc).

**Comment by AnandChowdhary** [comment]
You're right, but I think the problem is that this needs a server (e.g., a URL we can hit when a cron job runs correctly) but Upptime is totally dependent on GitHub Actions. Off the top of my head, I can't think of easy way to make this work, but happy to hear your ideas!
